extensions/wikia/TemplateClassification/TemplateClassificationController.class.php

```diff
@@ -49,4 +54,76 @@ public function getTemplateClassificationEditForm() {
 
 		$this->setVal( 'templateTypes', $templateTypes );
 	}
+
+	/**
+	 * Classify all templates with given category
+	 *
+	 * @requestParam string category category name
+	 * @requestParam string type template type name
+	 *
+	 * @throws ApiException
+	 * @throws \BadRequestApiException
+	 * @throws \InvalidParameterApiException
+	 * @throws \MWException
+	 * @throws \UnauthorizedException
+	 */
+	public function classifyTemplateByCategory() {
+		global $wgCityId;
+
+		$errors = [];
+		$category = $this->request->getVal( 'category', null );
+		$templateType = $this->request->getVal( 'type', null );
+		$userId = $this->wg->User->getId();
+
+		$this->validateRequestForBulkEdit( $this->wg->User, $this->request, $category, $templateType );
+
+		$templates = ( new Helper() )->getTemplatesByCategory( $category );
+		$utcs = new \UserTemplateClassificationService();
+
+		foreach ( $templates as $templateId => $templateTitle ) {
+			try {
+				$utcs->classifyTemplate($wgCityId, $templateId, $templateType, 'user', $userId );
+			} catch( ApiException $e ) {
+				$errors[] = \Title::newFromText( $templateTitle )->getText();
```

Comment:
Would it be worth validating the template type before launching into trying to classify every template (and getting an error for each one)?